src/include/sof/audio/module_adapter/library/native_system_service.h

```diff
@@ -0,0 +1,106 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ */
+/*! \file native_system_service.h */
+#ifndef NATIVE_SYSTEM_SERVICE_H
+#define NATIVE_SYSTEM_SERVICE_H
+#include "logger.h"
+#include "adsp_stddef.h"
+#include "adsp_error_code.h"
+#include <stdint.h>
+
+/*! \brief This struct defines the obfuscating type for notifications. */
+typedef struct _adsp_notification_handle {} *adsp_notification_handle;
+
+/*! \brief Defines parameters used by ADSP system during notification creation. */
+typedef struct _notification_params {
+	uint32_t type;              /*!< Notification type */
+	uint16_t user_val_1;        /*!< 16 bits user value available directly in IPC header
+				     * for some notifications
+				     */
+	uint32_t user_val_2;        /*!< 30 bits user value available directly in IPC header
+				     * for some notifications
+				     */
+	uint32_t max_payload_size;  /*!< Data size of payload (NotificationCreate updates this
+				     * value to max possible payload size)
+				     */
+	uint8_t *payload;           /*!< Pointer on the payload */
+} notification_params;
+
+/*! \brief Defines parameters used by ADSP system during Module Event notification creation. */
+typedef struct _module_event_notification {
+	uint32_t module_instance_id; /*!< Module ID (MS word) + Module Instance ID (LS word) */
+	uint32_t event_id;           /*!< Module specific event ID. */
+	uint32_t event_data_size;    /*!< Size of event_data array in bytes. May be set to 0
+				      * in case there is no data.
+				      */
+	uint32_t event_data[];       /*!< Optional event data (size set to 0 as it is optional
+				      * data)
+				      */
+} module_event_notification;
+
+/*! \brief Defines notification targets supported by ADSP system */
+typedef enum _notification_target {
+	DSP_TO_HOST = 1,  /*!< Notification target is HOST */
+	DSP_TO_ISH = 2    /*!< Notification target is ISH */
+} adsp_notification_target;
+
+/*! \brief Defines notification types supported by ADSP system */
+typedef enum _notification_type {
+	VOICE_COMMAND_NOTIFICATION = 4,	/*!< intel_adsp define corresponding to PHRASE_DETECTED
```

Comment:
Can we have a comment in each notification type to preserve ID for compatability. i.e. new types should start at value X

Reply:
Ditto, need a prefix.